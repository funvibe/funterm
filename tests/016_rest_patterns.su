lua.data = <<1, 2, 3, 4, 5, 6, 7, 8>>

# Binary rest (должен быть кратен 8 битам)
match lua.data {
    <<first:8, second:8, rest/binary>> -> {
        lua.print("First two:", first, second)
        lua.print("Binary rest length:", lua.len(rest))
    }
}

# Bitstring rest (любое количество битов)
lua.odd_bits = <<1:3, 2:5, 3:8, 4:4>>

match lua.odd_bits {
    <<a:3, b:5, rest/bitstring>> -> {
        lua.print("First parts:", a, b)
        lua.print("Bitstring rest bits:", lua.bit_size(rest))
    }
}

# Переменные размеры с rest
lua.packet = <<5:8, "Hello":5/binary, " World">>

match lua.packet {
    <<size:8, data:size/binary, rest/binary>> -> {
        lua.print("Size:", size)
        lua.print("Data:", data)
        lua.print("Rest:", rest)
    }
}