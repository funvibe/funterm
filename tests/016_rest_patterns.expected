First two: 1 2
Binary rest length: 6
First parts: 1 2
Bitstring rest bits: 12
Size: 5
Data: Hello
Rest:  World